pkg/operator/ceph/object/zone/controller.go

```diff
@@ -216,18 +216,24 @@ func (r *ReconcileObjectZone) createCephZone(zone *cephv1.CephObjectZone, realmN
 		return reconcile.Result{}, errors.Wrapf(err, "failed to parse `radosgw-admin zonegroup get` output")
 	}
 
-	masterArg := ""
-	// master zone does not exist yet for zone group
+	zoneIsMaster := false
 	if masterZone == "" {
-		masterArg = "--master"
+		zoneIsMaster = true
 	}
 
 	// create zone
 	_, err = object.RunAdminCommandNoRealm(objContext, "zone", "get", realmArg, zoneGroupArg, zoneArg)
 	if err != nil {
 		if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
 			logger.Debugf("ceph zone %q not found, running `radosgw-admin zone create`", zone.Name)
-			_, err := object.RunAdminCommandNoRealm(objContext, "zone", "create", realmArg, zoneGroupArg, zoneArg, masterArg)
+			if zoneIsMaster {
+				// master zone does not exist yet for zone group
+				masterArg := "--master"
+				_, err = object.RunAdminCommandNoRealm(objContext, "zone", "create", realmArg, zoneGroupArg, zoneArg, masterArg)
```

Comment:
What happens when a later reconcile is run? (for example, restart the operator). Is this idempotent to run the `zone create` for a master zone?

Reply:
If the reconcile is run, the zone & zonegroup CRs check to see if the zone/zonegroup already exist via a zone/zonegroup get. If the zone/zonegroup does not exist then it is created. 

That being said, it is not idempotent to run `zone/zonegroup create` with or without the --master flag. Is there anything I should be doing differently?

Reply:
Looks good, I see now that the master zone is only created if it doesn't exist already.